src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -195,6 +195,18 @@ public void run() {
                     callbackContext.success();
                 }
             });
+        } else if (GET_APPLICATION_ICON_BADGE_NUMBER.equals(action)) {
+            cordova.getThreadPool().execute(new Runnable() {
+                public void run() {
+                    Log.v(LOG_TAG, "getApplicationIconBadgeNumber");
+                    
+					int badgeCount = getApplicationIconBadgeNumber(getApplicationContext());
+					PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, badgeCount);
+					pluginResult.setKeepCallback(true);
```

Comment:
Why do you need to keep the callback around? Once you return the badge count you should be done.